### PR TITLE
Add glaive weapon support

### PR DESCRIPTION
**Pull request by JoShMiQueL**
- Update item_type.py
- Update common.py
- Update gen_data.py
- Update item_types.json

### CODE REVIEW

**Comment by JoShMiQueL** [comment]
I have made this pr to solve a problem I was personally having with this type of weapon, it is now working properly.

File: src/item/data/item_type.py [hunk 1]
```diff
@@ -21,6 +21,7 @@ class ItemType(Enum):
     OffHandTotem = "totem"
     Polearm = "polearm"
     Quarterstaff = "quarterstaff"
+    Glaive = "glaive"
```
Comment (chrisoro) [inline src/item/data/item_type.py]: sort alphabetical

File: src/tools/gen_data.py [hunk 2]
```diff
@@ -174,6 +174,7 @@ def main(d4data_dir: Path, companion_app_dir: Path):
             "OffHandTotem",
             "Polearm",
             "Quarterstaff",
+            "Glaive",
```
Comment (chrisoro) [inline src/tools/gen_data.py]: sort alphabetical

File: assets/lang/enUS/item_types.json [hunk 3]
```diff
@@ -19,6 +19,7 @@
     "OffHandTotem": "totem",
     "Polearm": "polearm",
     "Quarterstaff": "quarterstaff",
+    "Glaive": "glaive",
```
Comment (chrisoro) [inline assets/lang/enUS/item_types.json]: regen again, should be sorted alphabetical

**Review by chrisoro — APPROVED** [review]
Thank you!

**Pull request merged**